src/api/layer.js

```diff
@@ -282,17 +282,15 @@ export default class Layer {
     }
     async _styleChanged(style) {
         await this._context;
-        const originalPromise = this.requestData(style);
-        if (!originalPromise) {
-            // The previous stored metadata is still valid
+        return this.requestMetadata(style).then(metadata => {
+            if (metadata) {
```

Comment:
After the refactor I think this `if` is no longer needed

Reply:
After seeing with @jgoizueta we've seen that is needed unless we change the geojson source too. This is probably out scope then.

Reply:
With a Windshaft source we don't have any actual gain from that *optimization* (avoiding requesting the data if the metadata didn't change), so we could simplify the code by not making the distinction (metadata changed vs didn't). Because our Windshaft source the data already available.

But for other sources the optimization could be interesting, so I'd leave it as is.